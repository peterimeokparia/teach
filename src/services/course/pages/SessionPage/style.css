.session{
    position: absolute;
    background: white;
  }
  .session .content .sidebar {
    flex-basis: 300px;
    background: white;
    border-right: 0px solid #bba0ea;
    padding-left: 0px; 
  }
  .session .content .studentSessions .header{
    /* background-color:whitesmoke; */
    color: blue;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    padding: 0px 30px 5px 5px;
    text-align: center;
    text-transform: capitalize;
    margin-bottom: 25px;
    border-style: solid;
    border-radius: 3px;
    border-color: black;
  }
  .session .content .studentSessions .loginTime{ 
   margin-left: 150px;
  }
  .session .content .studentSessions .sessionHeader{ 
    margin-left: 75px;
   }
   .session .content .studentSessions .sessions{ 
    margin-left: 75px;
   }
   .session .content .studentSessions .totalsessions{ 
    margin-left: 155px;
   }
   .session .content .studentSessions .startdate{ 
    margin-left: 75px;
   }
   .session .content .studentSessions .enddate{ 
    margin-left: 15px;
   }
   .session .content .studentSessions .autorenew{ 
    margin-left: 25px;
   }
  .savedAnswersList{
    list-style: none;
    margin-top:  -10px;
    margin-left:  -320px;
    overflow: auto;
    height: 500px;
    width: 360px;
  }
  .session-item{
    text-transform: capitalize;
    padding-bottom: 20px;
    margin-bottom: 5px;
    margin-left: 30px;
    background: white;
  }
  .navlink-text {
    text-align: block;
    color: blue;
    padding-bottom: 20px;
  }
  .studentSessions{
    position: absolute;
    background: white;
    text-transform: capitalize;
    margin-left: -650px;
    width: 1000px;
    overflow: auto;
  }
  @media (min-width: 1281px) {
    
    /* CSS */
    
  }
  
  /* 
    ##Device = Laptops, Desktops
    ##Screen = B/w 1025px to 1280px
  */
  
  @media (min-width: 1025px) and (max-width: 1280px) {
    
    /* CSS */
    
  }
  
  /* 
    ##Device = Tablets, Ipads (portrait)
    ##Screen = B/w 768px to 1024px
  */
  
  @media (min-width: 768px) and (max-width: 1024px) {
    
    /* CSS */
    
  }
  
  /* 
    ##Device = Tablets, Ipads (landscape)
    ##Screen = B/w 768px to 1024px
  */
  
  @media (min-width: 768px) and (max-width: 1024px) and (orientation: landscape) {
    
    /* CSS */  
  }
  
  /* 
    ##Device = Low Resolution Tablets, Mobiles (Landscape)
    ##Screen = B/w 481px to 767px
  */
  
  @media (min-width: 481px) and (max-width: 767px) {    
    /* CSS */
    .studentSessions{
      position: absolute;
      background: white;
      text-transform: capitalize;
      margin-left: 10px;
      width: 320px;
      overflow: auto;
    }
  }
  /* 
    ##Device = Most of the Smartphones Mobiles (Portrait)
    ##Screen = B/w 320px to 479px
  */
  @media (min-width: 320px) and (max-width: 480px) {
    /* CSS */
    .session{
      /* position: relative; */
      background: white;
    }
    .session header {
      width: 20px;
    }
    .session .content .sidebar {
      flex-basis: 300px;
      background: white;
      border-right: 0px solid #bba0ea;
      padding-left: 0px; 
    }
    .savedAnswersList{
    list-style: none;
    margin-top:  -10px;
    margin-left:  -350px;
    overflow: auto;
    height: 300px;
    width: 220px;
  }
  .studentSessions{
    /* position: absolute; */
    /* position: relative; */
    background: white;
    text-transform: capitalize;
    margin-left: -80px;
    width: 250px;
    overflow: auto;
  }
  }