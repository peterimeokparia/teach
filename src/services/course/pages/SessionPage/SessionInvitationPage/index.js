import { 
    connect } from 'react-redux';
    
    import { 
    loadUsers } from 'services/course/actions/users';
    
    import useSessionHook from 'services/course/pages/SessionPage/hooks/useSessionHook';
    import './style.css';
    
    const SessionInvitationPage = ({
        user, 
        users }) => {
    
        let { 
            password,
            handleSubmit,
            handleOnChange,
         } = useSessionHook({ user, users });
       
    if ( users.filter( authenticateUser => authenticateUser?._id === user?._id)?.length === 1 ) {
        return ( <div>
                <form onSubmit={handleSubmit}>
                <label>
                    Password
                    <input 
                        required
                        value={password}
                        onChange={handleOnChange}
                        placeholder="Enter password to join video session"    
                    />
                </label>
                    <input type="submit" />
                </form>
        </div>)
    
        } else {
    
        return ( <div> <link to="/loginPage"> Login </link> and then join the session. </div>)
    }}
    
    const mapState = (state)   => {
        return {
            users: Object.values(state.users.users), 
            error: state.users.error,
            loading: state.users.loading  
        };
    }
    
    export default connect(mapState, {  loadUsers  })(SessionInvitationPage);