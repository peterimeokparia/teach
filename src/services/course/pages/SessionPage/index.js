import {
useEffect } from 'react';

import {
connect } from 'react-redux';

import { 
loadSessions,
saveSession } from 'services/course/actions/sessions';

import {
getSessionsByOperatorId } from 'services/course/selectors';

import {
Link } from '@reach/router';

import {
role } from 'services/course/helpers/PageHelpers';

import Roles from 'services/course/pages/components/Roles';
import ListItem from 'services/course/pages/components/ListItem';
import EditSessionComponent from './component/EditSessionComponent';
import './style.css';

const SessionPage = ({ 
    studentId,
    currentUser,
    courseId, 
    saveSession,
    loadSessions,
    sessions }) => {

    useEffect(() => {
        loadSessions();
    }, [ sessions?.length === 0]);

function onMatchListItem( match, listItem ) {
    if ( match ){
        console.log(`Selected: ${listItem}`);
    };
} 
return (    
    <div className="session">    
        {( sessions ) && <ListItem
                            // collection={Object.values(sessions?.filter(session => session?.userId === studentId && session?.courseId === courseId))}
                            collection={sessions?.filter(session => session?.userId === studentId && session?.courseId === courseId)}
                            onMatchListItem={onMatchListItem}
                            path={"student"}
                        >
                            {( selectedSession ) => (
                            < EditSessionComponent
                                studentId={studentId}
                                session={selectedSession}
                                className="lesson-item"
                                onSubmit={(session) => saveSession({...selectedSession, numberOfSessions: session?.numberOfSessions, totalNumberOfSessions: session?.totalNumberOfSessions  })}
                            >
                            { (edit, remove ) => (
                            <div className="content">      
                                <div className="studentSessions">
                                        <div className="header">
                                            {/* <span> Sessions </span> */}
                                            <span className="sessionHeader"> Type of Session </span>
                                            <span className="sessionHeader"> Number of Sessions</span>
                                            <span className="sessionHeader"> Total Number of Sessions </span>
                                            <span className="sessionHeader"> Start Date </span>
                                            <span className="sessionHeader"> End Date </span>
                                            <span className="sessionHeader"> Auto Renew </span>
                                        </div>
                                        <Link to={`student/${ studentId }/sessions/${selectedSession?._id}/courseId/${courseId}`}> <span title={selectedSession?._id} >{ selectedSession?.typeOfSession } </span> </Link> 
                                        <span className="sessions"> { selectedSession?.numberOfSessions } </span>
                                        <span className="totalsessions"> { selectedSession?.totalNumberOfSessions } </span>
                                        <span className="startdate"> { new Date( selectedSession?.startDate )?.toLocaleDateString('en-US') } </span>
                                        <span className="enddate"> { new Date( selectedSession?.endDate )?.toLocaleDateString('en-US') } </span>
                                        <span className="autorenew"> { selectedSession?.autoRenew.toString() } </span>
                                        <br></br>
                                        <div> 
                                <Roles
                                    role={currentUser?.role === role.Tutor }
                                >
                                    <button 
                                        className="edit-lesson-btn"
                                        onClick={() => { edit( selectedSession ); } }                                          
                                    > 
                                        Edit
                                    </button>
                                </Roles>
                                <Roles
                                    role={ currentUser?.role === role.Tutor }
                                >
                                    <span>
                                        <button
                                            className="delete-lesson-btn"
                                            onClick={() => { remove( selectedSession ); }}> 
                                            Delete 
                                        </button> 
                                    </span>
                                </Roles>
                                </div>  
                            </div>
                            </div>
                            )}
                        </EditSessionComponent> 
                            )
                        }
                    </ListItem>                    
                }
            </div>     
    );
};

const mapState = (state, ownProps)  => {
    return {
    currentUser: state?.users?.user,
    sessions: getSessionsByOperatorId(state, ownProps),
    lesson: state?.lessons?.lessons[ownProps?.courseId]
    // lesson: state?.lessons?.lessons[ownProps.lessonId]
    };
};

export default connect(mapState, { saveSession, loadSessions  } )(SessionPage);