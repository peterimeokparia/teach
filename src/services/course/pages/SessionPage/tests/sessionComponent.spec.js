import React from 'react';

import { 
Provider } from "react-redux";

import { 
applyMiddleware } from 'redux'; 

import { 
mockStoreObject } from 'services/course/api';

import SessionPage from 'services/course/pages/SessionPage';
import renderer from 'react-test-renderer';
import thunk from "redux-thunk"; 
import * as reactRedux from 'react-redux';
import configureStore from 'redux-mock-store';

import {
INCREMENT_SESSION_COUNT,
DECREMENT_SESSION_COUNT_FOR_PACKAGE_OPTIONS} from 'services/course/actions/sessions';

import {
incrementSessionCount,
decrementSessionCountForPackageOptions } from 'services/course/actions/sessions';

  const mockStore = configureStore([]);

jest.mock('../../../api');

describe('incrementSession',  () => {

  const sessionId = "SESSION5fab4846c2a96278c56381c9";
  const userId = "USER5fab4846c2a96278c56381c9";

  let user = {
      _id: userId
  };

  let sessions = {
      numberOfSessions: 1,
      _id: sessionId
  }; 
  
  it('should increment the number of sessions', async () => {

  const mockDispatch = jest.fn();

  await incrementSessionCount(
    sessions
  )(mockDispatch)

  console.log('new lesson', mockDispatch.mock.calls)
  expect(sessions.numberOfSessions).toBe(2);
});

it('should call 1 action', async () => {

    const mockDispatch = jest.fn();
  
    await incrementSessionCount(
      sessions
    )(mockDispatch)
  
    console.log('new session', mockDispatch.mock.calls[0][0])
    expect(mockDispatch.mock.calls.length).toBe(1);
    expect(mockDispatch.mock.calls[0][0]).toEqual({
        type: INCREMENT_SESSION_COUNT,
        payload: {
          _id: sessionId,
          numberOfSessions: 3
        }
    });   
  });
  
});

describe('decrementSessionCountForPackageOptions',  () => {

  const sessionId = "SESSION5fab4846c2a96278c56381c9";
  const userId = "USER5fab4846c2a96278c56381c9";

  let user = {
  _id: userId
  };

  let sessions = {
  numberOfSessions: 5,
  _id: sessionId
  };

it('should decrement the number of sessions', async () => {

  const mockDispatch = jest.fn();

  await decrementSessionCountForPackageOptions(
    sessions
  )(mockDispatch)

  console.log('new lesson', mockDispatch.mock.calls)
  expect(sessions.numberOfSessions).toBe(4);
});

it('should call 1 action', async () => {

  const mockDispatch = jest.fn();
  
  await decrementSessionCountForPackageOptions(
    sessions
  )(mockDispatch)

  console.log('new session', mockDispatch.mock.calls[0][0])
  expect(mockDispatch.mock.calls.length).toBe(1);
  expect(mockDispatch.mock.calls[0][0]).toEqual({
      type: DECREMENT_SESSION_COUNT_FOR_PACKAGE_OPTIONS,
      payload: {
        _id: sessionId,
        numberOfSessions: 3
      }
    });   
   });
});


describe('SessionPage', () =>  {  

  let  store = mockStore(mockStoreObject, applyMiddleware(thunk));

  const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');
 
  beforeEach(() => {
    useSelectorMock.mockClear();
    useDispatchMock.mockClear();
    jest.spyOn(Date.prototype, 'toLocaleTimeString')
      .mockImplementation(() => '12:00:00');
    jest.spyOn(Date.prototype, 'toLocaleDateString')
      .mockImplementation(() => 'AM 1/01/2021');
  });

  it('renders as expected', async () => {

    const dispatch = jest.fn();
    useDispatchMock.mockReturnValue(dispatch)

    const tree = renderer.create(
      <Provider store={store}>
        <SessionPage />
    </Provider>
    );
      expect(tree.toJSON()).toMatchSnapshot();
  });
});
  



