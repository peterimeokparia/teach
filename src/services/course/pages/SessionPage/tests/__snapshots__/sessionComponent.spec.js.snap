// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionPage renders as expected 1`] = `
<div
  className="session"
/>
`;
